### ISSUE TITLE
Negative operation for a boolean array is forbidden in NumPy

### ISSUE THREAD

**Issue by unnonouno** [cupy/cupy#692]
In NumPy negative operation for a boolean array is forbidden.
```
>>> -numpy.array([True])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: The numpy boolean negative, the `-` operator, is not supported, use the `~` operator or the logical_not function instead.
>>> -cupy.array([True])
array([False], dtype=bool)
```

**Comment by kmaehashi** [comment]
Related to this issue, NumPy raises DeprecationWarning for boolean subtract operation, which causes test failure (via pytest-warning).

```
>>> x = np.array([True])
>>> x - x
__main__:1: DeprecationWarning: numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logical_xor function instead.
array([False], dtype=bool)
>>> x = cp.array([True])
>>> x - x
array([False], dtype=bool)
```

Confirmed with NumPy 1.9.0.


Test result follows. Note that test is not deterministic (combinations are randomly chosen), so it may not always happen.

```
self = <cupy_tests.core_tests.test_ndarray_elementwise_op.TestArrayElementwiseOp testMethod=test_array_reversed_sub>

    def test_array_reversed_sub(self):
>       self.check_array_reversed_op(operator.sub)

self       = <cupy_tests.core_tests.test_ndarray_elementwise_op.TestArrayElementwiseOp testMethod=test_array_reversed_sub>

tests/cupy_tests/core_tests/test_ndarray_elementwise_op.py:458: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cupy/testing/helper.py:736: in test_func
    impl(self, *args, **kw_copy)
cupy/testing/helper.py:105: in test_func
    accept_error=accept_error)
cupy/testing/helper.py:49: in _check_cupy_numpy_error
    self.fail('Only numpy raises error\n\n' + numpy_tb)
E   AssertionError: Only numpy raises error
E   
E   Traceback (most recent call last):
E     File "/work/cupy/cupy/testing/helper.py", line 26, in _call_func
E       result = impl(self, *args, **kw)
E     File "/work/cupy/tests/cupy_tests/core_tests/test_ndarray_elementwise_op.py", line 452, in check_array_reversed_op
E       return op(a, b[::-1])
E   DeprecationWarning: numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logical_xor function instead.
```

**Comment by kmaehashi** [comment]
In NumPy 1.9.x, negative operation for a boolean array is permitted (but raises DeprecationWarning).

https://github.com/numpy/numpy/blob/maintenance/1.9.x/numpy/core/src/umath/ufunc_type_resolution.c#L386-L389

It was changed to TypeError in 1.13.0:

https://github.com/numpy/numpy/blob/maintenance/1.13.x/numpy/core/src/umath/ufunc_type_resolution.c#L384-L387

Maybe we need a handling policy when NumPy deprecates / removes interfaces.

Basically we should support union of feature sets of each supported NumPy versions.
For features that are expected to behave as the same way as NumPy, test cases must be separately written for each NumPy versions.
